ods_ci/tests/Tests/100__deploy/130__operators/140__serverless_operator/140__serverless_operator.robot

```diff
@@ -0,0 +1,111 @@
+
+
+*** Variables ***
+${RHOAI_APPNAME}  Red Hat OpenShift AI
+${RHOAI_OPERATOR_NAME}    Red Hat OpenShift AI
+${KSERVE_APPNAME}  Red Hat OpenShift Serverless
+${KSERVE_OPERATOR_NAME}    Red Hat OpenShift Serverless
+${KNATIVESERVING_NS}    knative-serving
+${RHOAI_OPERATOR_NS}    redhat-ods-operator
+
+
+*** Test Cases ***
+Validate DSC creates all Serverless CRs
+    [Documentation]  The purpose of this Test Case is to validate the creation
+    ...    of KServe Custom Resources
+    [Tags]  Operator    ODS-2600
+    Check And Install Operator In Openshift    ${RHOAI_APPNAME}    ${RHOAI_OPERATOR_NAME}
+    Check And Install Operator In Openshift    ${KSERVE_APPNAME}    ${KSERVE_OPERATOR_NAME}
+    Is Resource Present     KnativeServing    knative-serving     ${KNATIVESERVING_NS}
+    Check Status     oc get KnativeServing knative-serving -n ${KNATIVESERVING_NS} -o json | jq '.status.conditions[] | select(.type=="Ready") | .status'     KnativeServing    "True"    # robocop: disable
+    Is Resource Present     Gateway    knative-ingress-gateway     ${KNATIVESERVING_NS}
+    Is Resource Present     Gateway    knative-local-gateway     ${KNATIVESERVING_NS}
+    Is Resource Present     Service    knative-local-gateway     istio-system
+    Is Resource Present     deployment    controller     ${KNATIVESERVING_NS}
+    Check Pods Number    ${KNATIVESERVING_NS}    app.kubernetes.io/component=controller
+    ${pod_names}=    Get Pod Names    ${KNATIVESERVING_NS}    app.kubernetes.io/component=controller
+    Verify Containers Have Zero Restarts    ${pod_names}    ${KNATIVESERVING_NS}
+    ${podname}=    Get Pod Name   ${RHOAI_OPERATOR_NS}    label_selector=name=rhods-operator
+    Check For Errors On Operator Logs    ${podname}    ${RHOAI_OPERATOR_NS}
+    Check DSC Conditions    ${KNATIVESERVING_NS}    default-dsc
+
+
+*** Keywords ***
+Suite Setup
+    [Documentation]    Suite Setup
+    Set Library Search Order    SeleniumLibrary
+    RHOSi Setup
+
+Suite Teardown
+    [Documentation]    Suite Teardown
+    Close All Browsers
+    RHOSi Teardown
+
+Is Resource Present
+    [Documentation]    Check CR
+    [Arguments]       ${resource}     ${resource_name}    ${namespace}
+    ${rc}=     Run and Return Rc
+    ...  oc get ${resource} ${resource_name} -n ${namespace}
+    Should Be Equal    "${rc}"    "0"    msg=${resource} does not exist
+
+Check Status
+    [Documentation]    Check Resource Status
+    [Arguments]       ${oc_get}     ${resource}    ${expected_status}
+    ${status}=     Run
+    ...  ${oc_get}
+    Log    ${status}    console=True
+    Should Be Equal    ${status}    ${expected_status}   msg=${resource} is not in Ready status
+
+Check Pods Number
+    [Documentation]    Checks number of Pods
+    [Arguments]    ${namespace}    ${label_selector}
+    ${return_code}    ${output}    Run And Return Rc And Output    oc get pod -n ${namespace} -l app.kubernetes.io/component=controller | tail -n +2 | wc -l    # robocop: disable
```

Comment:
why not use juset ${label_selector} instead of app.kubernetes.io/component=controller

Reply:
@zdtsw Because this way Keyword is more generic, you can use any label

Reply:
We have keyword [Wait For Pods Numbers](https://github.com/red-hat-data-services/ods-ci/blob/master/ods_ci/tasks/Resources/RHODS_OLM/install/oc_install.robot#L191) - it may be moved to some common place and reused maybe?

Reply:
Replaced